include/CL/sycl/vec/detail/vec.hpp

```diff
@@ -90,8 +94,45 @@ class vec : public detail::small_array<DataType,
     return std::tuple_cat(flatten<V>(i)...);
   }
 
+protected:
+  __swizzled_base_vec__<DataType, 1>do_swizzle(int first) const {
+    return (*this)[first];
+  }
+
+  template<typename T, typename... Ts>
+  __swizzled_base_vec__<DataType, sizeof...(Ts) + 1>do_swizzle(T first, Ts... swizzleIndexes) const {
+    return __swizzled_base_vec__<DataType, sizeof...(Ts) + 1> {do_swizzle(first), do_swizzle(swizzleIndexes...) };
+  }
+
 public:
-  /// \todo To implement swizzles
+
+  /// Return the number of elements in the vector
+  auto get_count() const {
+    return NumElements;
+  }
+
+  /// Return the number of bytes
+  auto get_size() const {
+    return NumElements * sizeof(DataType);
+  }
+
+  template<typename convertT, rounding_mode roundingMode>
+  vec<convertT, NumElements> convert() const {
+    TRISYCL_UNIMPL;
+    return {};
+  };
+
+  template<typename asT> asT as() const {
+    TRISYCL_UNIMPL;
+    return {};
+  };
+
+  // Swizzle methods (see notes)
+  template<int... swizzleIndexs>
+  __swizzled_base_vec__<DataType, sizeof...(swizzleIndexs)> swizzle() const {
+    return do_swizzle(swizzleIndexs...);
```

Comment:
Probably we can remove `do_swizzle(swizzleIndexs...)` indirection and do the job directly here?

Reply:
okay I might have a better solution for this at least.